### PR TITLE
Feature/sts 15 integrate new key management in datasafe

### PR DESCRIPTION
**Pull request by max402**

### CODE REVIEW

File: security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java [hunk 1]
```diff
@@ -65,8 +79,12 @@ public String addToKeyStoreAndGetName(KeyStore ks, ProvidedKeyTemplate entry, Su
 
     @SneakyThrows
     private KeyStore generate(KeySet keySet, Supplier<char[]> defaultPassword, KeyMetadataOper useMetadataOper) {
-        KeyStore ks = KeyStore.getInstance("UBER"); // FIXME - BCFKS from provided load store config
-        ks.load(null);
+        KeyStore ks = KeyStore.getInstance(config.getType());
+        if ("BCFKS".equals(config.getType())) {
```
Comment (valb3r) [inline security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java]: It is better to have:
```
        KeyStore ks;
        ks.load(null);
        if ("BCFKS".equals(config.getType())) {
            ks = createBCFKSKeystore(config, ks);
        } else {
            ks = createKeyStoreByType(type);
        }
```
for readability

File: api/src/main/java/de/adorsys/keymanagement/api/persist/KeyStoreCreator.java [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 
+import de.adorsys.keymanagement.api.types.KeyStoreConfig;
 import de.adorsys.keymanagement.api.types.source.KeySet;
 
 import java.security.KeyStore;
 import java.util.function.Supplier;
 
 public interface KeyStoreCreator {
 
+    KeyStoreCreator withConfig(KeyStoreConfig config);
```
Comment (valb3r) [inline api/src/main/java/de/adorsys/keymanagement/api/persist/KeyStoreCreator.java]: Add JavaDoc

Comment (max402) [reply api/src/main/java/de/adorsys/keymanagement/api/persist/KeyStoreCreator.java]: ok

File: security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java (hunk 1 above)
Comment (max402) [reply security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java]: What should be inside createKeyStoreByType then?
We can't ks.load before initialize ks.
I've renamed createBCFKSKeystore to loadBCFKSKeystore.
Is it good to explicitly return ks even if it changes as a parameter?

Comment (valb3r) [reply security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java]: Yes - createKeyStoreByType should return keystore
```
 KeyStore ks;
        if ("BCFKS".equals(config.getType())) {
            ks = createBCFKSKeystore(config);
        } else {
            ks = createKeyStoreByType(type);
        }
```

Comment (max402) [reply security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java]: of course without changeable parameters is better

File: security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java [hunk 3]
```diff
@@ -19,10 +25,18 @@
 public class DefaultKeyStoreOperImpl implements KeyStoreOper {
 
     private final KeyMetadataOper metadataOper;
+    private KeyStoreConfig config;
```
Comment (valb3r) [inline security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java]: Make this field final

File: security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java [hunk 4]
```diff
@@ -19,10 +25,18 @@
...
     @Inject
     public DefaultKeyStoreOperImpl(KeyMetadataOper metadataOper) {
         this.metadataOper = metadataOper;
+        this.config = KeyStoreConfig.builder().build();
+    }
+
+    @Override
+    public DefaultKeyStoreOperImpl withConfig(KeyStoreConfig config) {
+        this.config = config;
+        return this;
```
Comment (valb3r) [inline security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java]: return new instance of DefaultKeyStoreOperImpl with proper config

File: security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java (hunk 3 above)
Comment (max402) [reply security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java]: I wanted set it withConfig(

**Pull request merged**